crypto/bio/bio_lib.c

```diff
@@ -784,3 +784,102 @@ void bio_cleanup(void)
     CRYPTO_THREAD_lock_free(bio_type_lock);
     bio_type_lock = NULL;
 }
+
+/* Internal variant of the below BIO_wait() not calling BIOerr() */
+static int bio_wait(BIO *bio, time_t max_time, unsigned long milliseconds)
+{
+    int fd;
+
+    if (max_time == 0)
+        return 1;
+
+    if (BIO_get_fd(bio, &fd) <= 0)
```

Comment:
Don't we want to do the ossl_sleep() in this case? This might happen with a non-socket based BIO.

Reply:
I'm not an export on sockets and file descriptors.
I thought that fds might be more generally applicable, but nevertheless I've changed it this way.